pkg/cpio/newc/newc.go

```diff
@@ -118,12 +119,21 @@ func (w *writer) WriteRecord(f cpio.Record) error {
 	buf := &bytes.Buffer{}
 	hdr := headerFromInfo(f.Info)
 	hdr.CRC = 0
+
+	// in newc, the FileSize is 0 for all but LNK and REG.
+	if f.Info.Mode&syscall.S_IFMT != syscall.S_IFREG &&
+		f.Info.Mode&syscall.S_IFMT != syscall.S_IFLNK {
+		hdr.FileSize = 0
```

Comment:
IOW it's not defined (not an error necessarily) for the headers here to have a non-zero size even if that makes no sense; it's an error to write a non-zero size for newc format however. I learned that the hard way.